keel-core/src/main/kotlin/com/netflix/spinnaker/keel/serialization/SubmittedEnvironmentDeserializer.kt

```diff
@@ -27,7 +30,7 @@ class SubmittedEnvironmentDeserializer : StdNodeBasedDeserializer<SubmittedEnvir
       val notifications: Set<NotificationConfig> = convert(root, "notifications") ?: emptySet()
       val locations: SubnetAwareLocations? = convert(root, "locations")
       val resources: Set<SubmittedResource<*>> = copy().run {
-        injectableValues = InjectableValues.Std(mapOf("locations" to locations))
+        injectableValues = InjectableLocations(locations)
```

Comment:
Fixed by creating my own implementation of this `InjectableValues` thing that can detect the desired type

Reply:
Do we lose any flexibility here for injecting additional (non-location) values in the same way?

Reply:
Yeah, we'd have to amend this as & when we do. If that's the case we may want an extension that allows for an optional conversion based on context.